server/src/main/java/org/opensearch/search/pipeline/StatefulSearchRequestProcessor.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.pipeline;
+
+import org.opensearch.action.search.SearchRequest;
+
+import java.util.Map;
+
+/**
+ * A specialization of {@link SearchRequestProcessor} that makes use of the request-scoped processor state.
+ */
+public interface StatefulSearchRequestProcessor extends SearchRequestProcessor {
```

Comment:
`SearchRequestProcessor` is not necessarily stateless, but I couldn't break the existing interface (that doesn't take state). One could just implement `SearchRequestProcessor` and implement the method that takes state, while throwing `UnsupportedOperationException` for the method that doesn't take state.

The sub-interfaces are just there to make it easier for developers. Extend the sub-interface and the compiler will force you to implement the method that takes state (since the default implementation from the parent has been cleared).